extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProcessor.java

```diff
@@ -38,4 +51,24 @@ void additionalProviders(BuildProducer<AdditionalBeanBuildItem> additionalBean,
                 .produce(new MessageBodyWriterBuildItem(JacksonMessageBodyWriter.class.getName(), Object.class.getName(),
                         Collections.singletonList(MediaType.APPLICATION_JSON)));
     }
+
+    @BuildStep
+    void registerForReflection(Optional<ResourceScanningResultBuildItem> resourceScanningResultBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+        if (!resourceScanningResultBuildItem.isPresent()) {
+            return;
+        }
+
+        Collection<ClassInfo> resourceClasses = resourceScanningResultBuildItem.get().getScannedResources().values();
+        Set<String> classesNeedingReflectionOnMethods = new HashSet<>();
+        for (ClassInfo resourceClass : resourceClasses) {
+            if (resourceClass.annotations().containsKey(JSON_VIEW)) {
+                classesNeedingReflectionOnMethods.add(resourceClass.name().toString());
```

Comment:
Sorry to be that guy but shouldn't we do this in the `quarkus-jackson` extension given it's a generic Jackson annotation?

It could be used in other extensions then.

And I would do it for all `@JsonView` annotations in the classpath.

Reply:
No, because this reflection is only needed in order to make `resteasy-reactive` work. Normally RESTEasy reactive does **not** have to perform reflection on the resource method, but in this case (and in other that we will surely find later on) it does.

Reply:
Maybe I'm missing something but if we do need reflection for the class referenced in JsonView, I think it's a general requirement.

I'm not sure though if you added reflection for that?

Reply:
We are not doing reflection for the class referenced in the annotation, but for the class that contains the resource method in quesion.
Perhaps the variables names are not clear enough?

Reply:
They are clear but I'm still puzzled on how it can work without having the class in `@JsonView` registered for reflection.

Reply:
Why do you think the value needs to be registered? Jackson almost certainly doesn't need to do any reflection on this class. It probably just has to call `isAssignable` on the class